src/tests/unittests/validation/BufferValidationTests.cpp

```diff
@@ -317,3 +317,29 @@ TEST_F(BufferValidationTest, SetSubDataWrongUsage) {
     uint32_t foo = 0;
     ASSERT_DEVICE_ERROR(buf.SetSubData(0, 1, &foo));
 }
+
+// Test that command buffer submit changes buffer usage
```

Comment:
nit: Given all the tiny details of usage validation, I suggest we make a UsageValidationTests file.